dialoptions.go

```diff
@@ -377,7 +382,15 @@ func WithInsecure() DialOption {
 // later release.
 func WithNoProxy() DialOption {
 	return newFuncDialOption(func(o *dialOptions) {
-		o.copts.UseProxy = false
+		o.UseProxy = false
+	})
+}
+
+// WithTargetResolutionEnabled returns a DialOption which enables target
+// resolution on client. This is ignored if WithNoProxy is used.
+func WithTargetResolutionEnabled() DialOption {
```

Comment:
Please mark this function as experimental. We keep new APIs experimental initially so they can be changed based on user feedback.
```
//
// # Experimental
//
// Notice: This API is EXPERIMENTAL and may be changed or removed in a
// later release.
```

Reply:
Done.